pymongo/client_session.py

```diff
@@ -539,6 +558,14 @@ def advance_operation_time(self, operation_time):
                             "of bson.timestamp.Timestamp")
         self._advance_operation_time(operation_time)
 
+    def _process_response(self, reply):
+        """Process a response to a command that was run with this session."""
+        self._advance_cluster_time(reply.get('$clusterTime'))
+        self._advance_operation_time(reply.get("operationTime"))
+        recovery_token = reply.get('recoveryToken')
+        if recovery_token and self._in_transaction:
```

Comment:
As far as I know, recoveryToken is only returned from commands run inside a transaction. Maybe this would be clearer:
```python
        self._advance_operation_time(reply.get("operationTime"))
        if self._in_transaction:
            recovery_token = reply.get('recoveryToken')
            if recovery_token:
                self._transaction.recovery_token = recovery_token
```